### PR TITLE
Upgrade to polkadot v0.9.42

### LINKED ISSUE #320: upgrade dependencies to 0.9.42

### PR DESCRIPTION
**Pull request by haerdib**
Update to polkadot-v0.9.42 and substrate-api-client to tag v0.10.0.

Follow up PR of  #325

closes #320 

FYI: with the new api-client the function `submit_and_watch_extrinsic_until_success` is available. I believe that should open quite some features for api-client (e.g. it is known when an xt is included in block, no need to wait for 3 blocks to pass and it allows ascertaining that the extrinsic has actually been executed).

### CODE REVIEW

**Comment by haerdib** [comment]
CI passed for `test_bot_community` but it spams the following error:

```
[2023-06-06T10:23:30Z ERROR encointer_client_notee] ExtrinsicFailed: EventDetails { phase: ApplyExtrinsic(1), index: 0, all_bytes: [4, 0, 1, 0, 0, 0, 0, 1, 3, 61, 16, 0, 0, 0, 7, 112, 130, 49, 28, 1, 0, 0, 0, 0], start_idx: 1, fields_start_idx: 8, fields_end_idx: 23, end_idx: 24, metadata: Metadata { metadata: RuntimeMetadataV14 { types: PortableRegistry { types: [PortableType { id: 0, ty: Type { path: Path { segments: ["sp_core", "crypto", "AccountId32"] }, type_params: [], type_def: Composite(TypeDefComposite { fields: [Field { name: None, ty: UntrackedSymbol { id: 1, marker: PhantomData<fn() -> core::any::TypeId> }, type_name: Some("[u8; 32]"), docs: [] }] }), docs: [] } }, PortableType { id: 1, ty: Type { path: Path { segments: [] }, type_params: [], type_def: Array(TypeDefArray { len: 32, type_param: UntrackedSymbol { id: 2, marker: PhantomData<fn() -> core::any::TypeId> } }), docs: [] } }, PortableType { id: 2, ty: Type { path: Path { segments: [] }, type_params: [], type_def: Primitive(U8), docs: [] } }, PortableType { id: 3, ty: Type { path: Path { segments: ["frame_system", "AccountInfo"] }, type_params: [TypeParameter { name: "Index", ty: Some(UntrackedSymbol { id: 4, marker: PhantomData<fn() -> core::any::TypeId> }) }, TypeParameter { name: "AccountData", ty: Some(UntrackedSymbol { id: 5, marker: PhantomData<fn() -> core::any::TypeId> }) }], type_def: Composite(TypeDefComposite { fields: [Field { name: Some("nonce"), ty: UntrackedSymbol { id: 4, marker: PhantomData<fn() -> core::any::TypeId> }, type_name: Some("Index"), docs: [] }, Field { name: Some("consumers"), ty: UntrackedSymbol { id: 4, marker: PhantomData<fn() -> core::any::TypeId> }, type_name: Some("RefCount"), docs: [] }, Field { name: Some("providers"), ty: UntrackedSymbol { id: 4, marker: PhantomData<fn() -> core::any::TypeId> }, type_name: Some("RefCount"), docs: [] }, Field { name: Some("sufficients"), ty: UntrackedSymbol { id: 4, marker: PhantomData<fn() -> core::any::TypeId> }, type_name: Some("RefCount"), docs: [] }, Field { name: Some("data"), ty: UntrackedSymbol { id: 5, marker: PhantomData<fn() -> core::any::TypeId> }, type_name: Some("AccountData"), docs: [] }] }), docs: [] } }, PortableType { 
```

I checked previous PR and the error was there as well. Is that a known issue?

**Comment by clangenb** [comment]
Hmmm, yes, I think this error is familiar. As our bot-community script is insanely stupid, it does sometimes send failing txs. Important is that the tests in that script pass.

File: client/src/main.rs [hunk 1]
```diff
@@ -1837,41 +1863,21 @@ fn get_community_identifiers(
...
-		"jsonrpc": "2.0",
-		"id": "1",
-	});
-
-	let n = api.get_request(req).unwrap().expect("Could not find any business offerings...");
-	Some(serde_json::from_str(&n).unwrap())
+	api.client()
+		.request("encointer_bazaarGetOfferings", rpc_params![cid])
+		.expect("Could not find any business offerings...")
 }
```
Comment (clangenb) [inline client/src/main.rs:1881]: Nooiiice!

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -362,6 +357,10 @@ impl pallet_balances::Config for Runtime {
 	type ExistentialDeposit = ConstU128<EXISTENTIAL_DEPOSIT>;
 	type AccountStore = System;
 	type WeightInfo = pallet_balances::weights::SubstrateWeight<Runtime>;
+	type HoldIdentifier = ();
+	type FreezeIdentifier = ();
+	type MaxHolds = ();
+	type MaxFreezes = ();
 }
```
Comment (clangenb) [inline runtime/src/lib.rs]: We have found out meanwhile that we need to set them to `Const<0>`. :)

**Review by clangenb — COMMENTED** [review]
Beautiful, thanks for helping out here, this really made a difference! There is only one detail, that we need to fix. :)

File: runtime/src/lib.rs (hunk 2 above)
Comment (haerdib) [reply runtime/src/lib.rs]: All of them? Including the ones previously defined ? (MaxReserves, DustRemoval, ... )

Comment (clangenb) [reply runtime/src/lib.rs]: nope just `MaxHolds` and `MaxFreezes`

Comment (haerdib) [reply runtime/src/lib.rs]: Done

**Review by clangenb — APPROVED** [review]
Nice, thanks for the fixes!

**Pull request merged**